pkg/ruler/compat.go

```diff
@@ -96,6 +100,19 @@ func engineQueryFunc(engine *logql.Engine, overrides RulesLimits, checker readyC
 	})
 }
 
+// applyJitter sleeps up to a maximum of EvaluationJitter seconds to spread out evaluations to prevent thundering herd
+func applyJitter(rng *rand.Rand, jitter time.Duration, logger log.Logger) float64 {
```

Comment:
Nits on naming:
- The comment refers to the `jitter` argument as `EvaluationJitter`.
- I'd rename the `jitter` argument to `maxJitter`.
- I'd rename the `j` variable to `jitter`.